clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -830,6 +845,10 @@ class SyclKernelFieldChecker
             << 1 << FieldTy;
     }
   }
+  void handleKernelObject(CXXRecordDecl *KernelObject,
+                          QualType KernelType) final {
+    // Do we need any diagnostics for Kernel Object?
```

Comment:
Since kernel object can be defined not only as lambda but as a callable user's type, we could validate that the kernel object is trivially copy constructible (or satisfies standard layout requirement). It is required by the spec (section 4.8.9.1)